### PR TITLE
Eliminate all credo issues

### PR DESCRIPTION
**Pull request by mbramson**
Eliminates all remaining credo issues and adds a credo configuration file.

### CODE REVIEW

File: .credo.exs [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# This file contains the configuration for Credo and you are probably reading
```
Comment (mbramson) [inline .credo.exs:1]: This file was added via `mix credo gen.config`.

File: .credo.exs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      # to the tuple.
+      #
+      # To disable a check put `false` as second element:
+      #
+      #     {Credo.Check.Design.DuplicatedCode, false}
+      #
+      checks: [
+        {Credo.Check.Consistency.ExceptionNames},
+        {Credo.Check.Consistency.LineEndings},
+        {Credo.Check.Consistency.MultiAliasImportRequireUse, false},
```
Comment (mbramson) [inline .credo.exs:52]: This was my only change to the generated configuration. This makes credo not check for inconsistencies in import style. This check enforces that all imports are either

```
alias Pairmotron.User
alias Pairmotron.Pair
```
or 
```
alias Pairmotron.{User, Pair}
```
This check is a bit buggy, as I eliminated most cases of this inconsistency until I came across an issue where it seems to do the actual check after resolving the `__using__` macro. This is particularly an issue in `Pairmotron.Web`.

After some pontification, I'm not sure that I agree with the check in the first place. While consistency is good, sometimes you'd like to alias many things, and they're not all the same type of thing despite being imported from the same module. As an example, `Pairmotron.Types` defines some types used only for TypeSpecs, while `Pairmotron.{User, Pair, Group}` are all Ecto schemas. It's a bit misleading to lump them all together as `Pairmotron.{User, Pair, Group, Types}`.

File: lib/pairmotron.ex [hunk 3]
```diff
@@ -4,6 +4,7 @@ defmodule Pairmotron do
 
   # See http://elixir-lang.org/docs/stable/elixir/Application.html
   # for more information on OTP Applications
+  @spec start(atom(), map()) :: Supervisor.on_start
```
Comment (mbramson) [inline lib/pairmotron.ex:7]: You could definitely make an argument that these didn't need Typespecs, but they don't hurt, and it was fun tracking them down.

**Review by mbramson — COMMENTED** [review]
Thoughts and commentary.

File: lib/pairmotron.ex (hunk 3 above)
Comment (ericworkman) [reply lib/pairmotron.ex:7]: `Supervisor.on_start` neato

**Review by ericworkman — APPROVED** [review]
Awesome! LGMT!

**Pull request merged**